assembly/net/index.ts

```diff
@@ -237,337 +188,407 @@ declare function tcp_stream_deserialize(tcp_stream: u32): u32;
 // @ts-ignore: valid decorator
 @external("lunatic", "tcp_bind")
 declare function tcp_bind(
-  addr_ptr: usize,// *const u8,
   addr_len: usize,
+  addr_ptr: usize,// *const u8,
   port: u16,
+  flow_info: u32,
+  scope_id: u32,
   listener_id: usize, //*mut u32,
-): TCPErrorCode;
-
-/**
- * Close a tcp server.
- *
- * @param {u32} listener - The TCPServer id that should be closed.
- */
-// @ts-ignore: valid decorator
-@external("lunatic", "close_tcp_listener")
-declare function close_tcp_listener(listener: u32): void;
+): err_code;
 
 /**
  * Block the current thread to accept a socket from the TCPServer.
  *
  * @param {u32} listener - The listener.
- * @param {usize} tcp_socket - A pointer to a u32 that will contain the socket id.
- * @returns {TCPAcceptResult} The result of accepting a tcp socket.
+ * @param {usize} tcp_socket - A pointer to a u64 that will contain the socket id.
+ * @param {usize} address_ptr - A pointer to a u64 that will contain a dns iterator.
+ * @returns {err_code} The result of accepting a tcp socket.
  */
 // @ts-ignore: valid decorator
 @external("lunatic", "tcp_accept")
 declare function tcp_accept(
   listener: u32,
-  tcp_socket: usize, //*mut u32
-): TCPErrorCode;
+  tcp_socket: usize, //*mut u64
+  address_ptr: usize, //*mut u64
+): err_code;
 
 /**
- * Serialize a TCPServer.
+ * Initiate a TCP connection.
  *
- * @param {u32} tcp_listener - The listener to serialize
- * @returns {u32} The serialized TCPServer handle.
+ * @param {usize} addr_ptr - A pointer to an array of bytes.
+ * @param {usize} addr_len - A pointer to a usize that describes the length of the IP address.
+ * @param {u16} port - The port.
+ * @param {u32} flow_info - The flow info. (ipv6)
+ * @param {u32} scope_id - The scope id. (ipv6)
+ * @param {usize} listener_id - A pointer to a u32 where the listener_id will be written if
+ * the connection was successful, or the error id if it failed.
+ * @returns {TCPErrorCode} The result of initiating a TCP connection.
  */
 // @ts-ignore: valid decorator
-@external("lunatic", "tcp_listener_serialize")
-declare function tcp_listener_serialize(tcp_listener: u32): u32;
+@external("lunatic", "tcp_connect")
+declare function tcp_connect(
+  addr_ptr: usize, // *const u8,
```

Comment:
I think the order of `addr_ptr` and `addr_len` was reversed. https://github.com/lunatic-solutions/rust-lib/blob/main/src/host_api.rs#L46

Reply:
There is no more `addr_len` (4 or 16), only `addr_type` (4 or 6, refering to the IPv4 or IPv6 (16 bytes) formats)

Reply:
Hey guys. Is this an `u32` then?